### PR TITLE
Update how-to-release instructions

### PR DESCRIPTION
**Pull request by doscott**
I have tried to update the instructions to achieve two things:
 - a clear top to bottom sequence
 - a bit more detail on the how and why

I am not a markdown user, so I used ghostwriter.

### CODE REVIEW

File: docs/contributing/how-to-release.md [hunk 1]
```diff
@@ -182,22 +192,28 @@ Submit to https://isocpp.org/blog/suggest (need to be logged in to do it).
 
 For major releases, submit the announcement to https://slashdot.org/submission
 
-## Version Updates
+## Post-Release Steps
 
-Trac: mark the milestone corresponding to the release as completed and add a
+* Trac: mark the milestone corresponding to the release as completed and add a
```
Comment (vadz) [inline docs/contributing/how-to-release.md:197]: The change of `-` to `*` here and to `+` below seems weird, was this intentional?

File: docs/contributing/how-to-release.md [hunk 2]
```diff
@@ -182,22 +192,28 @@ Submit to https://isocpp.org/blog/suggest (need to be logged in to do it).
...
 in "Tickets to be fixed for the next release".
 
-Run `misc/scripts/inc_release` to increment micro version, i.e. replace x.y.z
-with x.y.z+1 (minor or major versions updates require manual intervention)
-and rerun both `bakefile_gen` and `autoconf` afterwards to update the version
-in the generated files too.
++ Run `misc/scripts/inc_release` to increment micro version, i.e. replace x.y.z
+with x.y.z+1.
+
++ Update the C:R:A settings in build/tools/version.bkl to X:1:X. Then from the build/bakesfiles directory run
```
Comment (vadz) [inline docs/contributing/how-to-release.md:210]:
```suggestion
+ Update the C:R:A settings in build/tools/version.bkl to C:R+1:A. Then from the build/bakesfiles directory run
```

File: docs/contributing/how-to-release.md [hunk 3]
```diff
@@ -67,7 +69,15 @@ and then run it using the new DLLs.
     * Update `docs/msw/binaries.md`: at least the version, but possibly also
       the list of supported compilers.
 
-2. Commit the changes and tag the release using your GPG key:
+2. For micro relases, review the changelog to determine if any ABI changes have been made. If so, or if this is a major or minor release, update the C:R:A settings in build/tools/version.bkl. Then from the build/bakesfiles directory run
```
Comment (vadz) [inline docs/contributing/how-to-release.md:72]:
```suggestion
2. For micro relases, review the changelog to determine if any ABI changes have been made. If so, or if this is a major or minor release, update the C:R:A settings in `build/tools/version.bkl`. Then from the `build/bakesfiles` directory run
```
Just to do it consistently, as it's already done for the other file/directory names in this file.

**Review by vadz — COMMENTED** [review]
Thanks, this looks good to me, but I have a couple of minor nitpicks and would also prefer to wrap the long lines to keep them under 80 (maybe even 76) columns as this file should remain readable in the terminal too.

**Comment by doscott** [comment]
I am push an update to apply your suggestions, and I have gone through the entire file to keep line lengths below 78. There are two long lines I didn't touch (line numbers in push to follow):
[180]    $ PATH="$HOME/doxygen/bin:$PATH" WX_SKIP_DOXYGEN_VERSION_CHECK=1 nice -n 15 ./regen.sh php
[245]     - Windows SDK 6.1, 7.1 (required for x64 builds for Visual Studio 2008, 2010)

**Comment by vadz** [comment]
Sorry, I should have really been more precise, I only meant to ask you to wrap a couple of _new_ long lines, I really don't think it's a good idea to reformat the entire file as this would mess up its history. So finally I'm applying only the actual additions, but not the reformatting. Sorry for wasting your time.

**Pull request closed without merging**